Add option `ignoreAtRules: []` for at-rule-empty-line-before

Nirazul opened the pull request:
I thought I'd give it a try and implement #1377 as I've been working on the rule, anyway.

jeddy3 commented on src/rules/at-rule-empty-line-before/index.js:
```diff
@@ -6,6 +8,7 @@ import {
   ruleMessages,
   validateOptions,
 } from "../../utils"
+import { cssStatementIsIgnoredAtRule } from "../block-opening-brace-space-before"
```
I think we should pull this out of `block-opening-brace-space-before` and into a util called `optionsHaveIgnoredAtRule(options, atRule)` so that it matches [the existing](https://github.com/stylelint/stylelint/blob/master/src/utils/optionsHaveIgnored.js) options utils. @davidtheclark What do you think?

davidtheclark replied:
Yes, we don't want these rules depending on each other. Would be better with a util.

jeddy3 commented on src/rules/at-rule-empty-line-before/__tests__/index.js:
```diff
@@ -380,3 +380,163 @@ testRule(rule, mergeTestDescriptions(sharedNeverTests, {
+
+testRule(rule, {
+  ruleName,
+  config: [ "always", { ignoreAtRules: "/fo/" } ],
```
Can we use `/el/` (matching `else`) here to avoid the ellipses in the code below?

jeddy3 commented:
@Nirazul Thanks for tackling this! I've one request for the tests and another about where that options function goes. We should wait for @davidtheclark to feedback on the latter before doing anything.

Can you also update the CHANGELOG please?

Nirazul commented:
I've changed things according to your requests.

Strangely, I had to remove a line in [selector-type-case/README.md](https://github.com/stylelint/stylelint/blob/master/src/rules/selector-type-case/README.md) that made the ci tests fail...

I've also removed a duplicated line in the CHANGELOG Head, that seemed to have been duplicated during the last merge.

davidtheclark commented:
Great work! I probably messed up the changelog on a recent confusing rebase. This LGTM.

jeddy3 commented:
@Nirazul Good stuff!
